phoenix-core/src/main/java/org/apache/phoenix/util/SchemaUtil.java

```diff
@@ -1238,6 +1238,78 @@ public static String getNormalizedColumnName(ColumnParseNode columnParseNode) {
         return columnParseNode.getName();
     }
 
+    public static String getFullTableNameWithQuotes(String schemaName, String tableName,
+    boolean schemaNameCaseSensitive, boolean tableNameCaseSensitive) {
+        String fullTableName;
+
+        if (tableNameCaseSensitive) {
+            fullTableName = "\"" + tableName + "\"";
+        } else {
+            fullTableName = tableName;
+        }
+
+        if(schemaName != null && schemaName.length() != 0) {
+            if (schemaNameCaseSensitive) {
+                fullTableName = "\"" + schemaName + "\"" + QueryConstants.NAME_SEPARATOR + fullTableName;
+            } else {
+                fullTableName = schemaName + QueryConstants.NAME_SEPARATOR + fullTableName;
+            }
+        }
+        return fullTableName;
+    }
+
+    public static String getFullTableNameWithQuotes(String schemaName, String tableName) {
+        return getFullTableNameWithQuotes(schemaName, tableName,
+                quotesNeededForSchema(schemaName), quotesNeededForTable(tableName));
+    }
+
+    private static boolean quotesNeededForSchema(String name) {
+        if (Strings.isNullOrEmpty(name) || name.equals(QueryConstants.DEFAULT_COLUMN_FAMILY)) {
+            return false;
+        }
+        return quotesNeededForTable(name);
+    }
+
+    private static boolean quotesNeededForColumn(String name) {
+        if (!name.equals("_INDEX_ID") && name.startsWith("_")) {
+            return true;
+        }
+        return isQuotesNeeded(name) || containsLowerCase(name);
+    }
+
+    private static boolean quotesNeededForTable(String name) {
+        if (name.startsWith("_")) {
+            return true;
+        }
+        return isQuotesNeeded(name) || containsLowerCase(name);
+    }
+
+    public static String formatSchemaName(String name) {
+        if (quotesNeededForSchema(name)) {
```

Comment:
Does the leading _ issue not apply here ?

Reply:
It does, in `quotesNeededForSchema` we check for some extra schema specific stuff but I call the `quotesNeededForTable` there.